enrichments/trace/internal/elastic/span_test.go

```diff
@@ -0,0 +1,447 @@
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetName: "testsvc",
+			},
+		},
+		{
+			name: "http_span_basic",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutInt(
+					semconv.AttributeHTTPResponseStatusCode,
+					http.StatusOK,
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "http",
+				AttributeServiceTargetName: "testsvc",
+			},
+		},
+		{
+			name: "http_span_full_url",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutInt(
+					semconv.AttributeHTTPResponseStatusCode,
+					http.StatusOK,
+				)
+				span.Attributes().PutStr(
+					semconv.AttributeURLFull,
+					"https://www.foo.bar:443/search?q=OpenTelemetry#SemConv",
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "http",
+				AttributeServiceTargetName: "www.foo.bar:443",
+			},
+		},
+		{
+			name: "http_span_no_full_url",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutInt(
+					semconv.AttributeHTTPResponseStatusCode,
+					http.StatusOK,
+				)
+				span.Attributes().PutStr(semconv.AttributeURLDomain, "www.foo.bar")
+				span.Attributes().PutInt(semconv.AttributeURLPort, 443)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "http",
+				AttributeServiceTargetName: "www.foo.bar:443",
+			},
+		},
+		{
+			name: "rpc_span_grpc",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutInt(
+					semconv.AttributeRPCGRPCStatusCode,
+					int64(codes.OK),
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "grpc",
+				AttributeServiceTargetName: "testsvc",
+			},
+		},
+		{
+			name: "rpc_span_system",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
```

Comment:
```suggestion
```

Does this comment belong here? It seems that `peer.service` is honoured, per the enriched attributes.

Reply:
Copy paste fiasco 😮‍💨